From 5be621b9a129b065cd546f9cf3d535f25dcd567a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 01 Aug 2021 12:46:05 +0300
Subject: [PATCH] [SW-51308] gaudi: security: allow pcie gw access

For the host to be able to issue transactions through the GW, the
protection bits must be reset and allow read and write to the GW
registers.
Rework the PCIe WRAP protection bits and open the GW registers.

Change-Id: I85940bcf80a0934bad7659ff0a29156957a0a9ba
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index f23f8bb..c75ae12 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -863,7 +863,18 @@
 		PROT_BIT(mmPCIE_WRAP_RR_ELBI_RD_SEC_REG_CTRL));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0xF80 */
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0x300 */
+	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE);
+
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x300 - 0x380 */
+	mask = PROT_BIT(mmPCIE_WRAP_LBW_GW_ADDR_0 - sizeof(u32));
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
+
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x400 */
+	mask = ~PROT_BIT(mmPCIE_WRAP_LBW_GW_STATUS_7);
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
+
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x400 - 0xF80 */
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX - (prot_base - base));
 }
 
